src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -725,8 +725,7 @@ private void computeUpstreamNodesWhenWorkspaceModified(WorkspacesModificationEve
             if (args.Id == this.Guid)
             {
                 this.workspaceLoaded = true;
-                var frozenNodes = nodes.Where(x => x.isFrozenExplicitly).ToList();
-                frozenNodes.ForEach(z => z.ComputeUpstreamOnDownstreamNodes());
+                this.ComputeUpstreamCacheForEntireGraph();
```

Comment:
But is there any specific reason to compute the cache for the entire graph? My guess is we can compute for only those frozen nodes, if we are not going to use this function for other reasons. we can talk tomorrow.